internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -140,30 +143,40 @@ func (u *ArtifactBucketUploader) uploadAssets(assets []asset) error {
 	return g.Wait()
 }
 
-// uploadAssetMappingFile uploads a JSON file to u.AssetMappingPath containing
+// uploadAssetMappingFile uploads a JSON file to u.AssetMappingDir containing
 // the current location of each file in the artifact bucket and the desired location
 // of the file in the destination bucket. It has the format:
 //
-//	{
+//	[{
 //	  "path": "local-assets/12345asdf",
 //	  "destPath": "index.html"
-//	}
-func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) error {
-	// stable output
+//	}]
+//
+// The uploaded path of the file is returned along with an error, if any.
+func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) (string, error) {
 	sort.Slice(assets, func(i, j int) bool {
```

Comment:
Will this ever result in a equal comparison (I assume no - it probably would never make sense for two files to be uploaded to the same bucket and the same path, no matter how this package evolves 🤔 💭  ..)? If so, just in case, maybe we want to use https://pkg.go.dev/sort#SliceStable

Reply:
it _is_ possible that two mappings are the exact same - though likely not what the user is intending to configure. e.g., two uploads: one of folder `./static` and one of file `./static/index.html` with `dest: ./static` would result in _two_ `index.html hash` -> `/static/index.html` entries in assets

i think the right approach here is probably just to remove any duplicate elements instead of using slicestable - thoughts?

Reply:
> e.g., two uploads: one of folder ./static and one of file ./static/index.html with dest: ./static would result in two index.html hash -> /static/index.html entries in assets

Makes sense! In this case, the user could have just not specified `./static/index.html` right? If we don't remove the duplicated entries, it wouldn't have caused issues because we basically just upload the same thing twice. In this case, then

> i think the right approach here is probably just to remove any duplicate elements instead of using slicestable


Agreed!

Reply:
> the user could have just not specified ./static/index.html right?

yeah, exactly! 👍